### PR TITLE
doc: Add Ruby pg guide

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

This adds a Ruby guide that follows the format of the [Node.JS](https://materialize.com/docs/guides/node-js/) and [PHP](https://materialize.com/docs/guides/php-pdo/) and Golang guides that we already have.

### CODE REVIEW

File: doc/user/content/guides/ruby.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+```
+
+For more information, see [`CREATE SOURCE`](/sql/create-source/).
+
+### Create a view from Ruby
+
+```ruby
+require 'pg'
+require 'pp'
+conn = PG.connect("127.0.0.1", 6875, '', '', "materialize", "materialize", "materialize")
```
Comment (ruf-io) [inline doc/user/content/guides/ruby.md:169]: switch to hash style

**Review by ruf-io — CHANGES_REQUESTED** [review]
Two high-level questions/requests:

1. Is it necessary to include `pp`? 
2. Switch to hash style for connect syntax

File: doc/user/content/guides/ruby.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+## Stream
+
+To take full advantage of incrementally updated materialized views from a Ruby application, instead of [querying](#query) Materialize for the state of a view at a point in time, use [a `TAIL` statement](/sql/tail/) to request a stream of updates as the view changes.
+
+To read a stream of updates from an existing materialized view, open a long-lived transaction with `BEGIN` and use [`TAIL` with `FETCH`](/sql/tail/#tailing-with-fetch) to repeatedly fetch all changes to the view since the last query.
+
+```ruby
+require 'pg'
+require 'pp'
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply doc/user/content/guides/ruby.md]: Yep good point! Better to keep it consistent. 

Initially, I used `pp` to inspect the `PG::Result class`. As if you do `puts result` you would get `#<PG::Result:0x000000015a100b48>` which does not give you much information, where as with `pp` you would get information about the status and etc: `#<PG::Result:0x0000000125087980 status=PGRES_COMMAND_OK ntuples=0 nfields=0 cmd_tuples=0>`.

But I could just have used the `.inspect` method with `puts` as follows: `puts result.inspect` to do the same.

Just submitted the changes so it is consistent throughout the doc.

File: doc/user/content/guides/ruby.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
+---
+title: "Ruby and Materialize"
+description: "Use Ruby to connect, insert, manage, query and stream from Materialize."
+weight:
+menu:
+  main:
+    parent: guides
+---
+
+Materialize is **PostgreSQL-compatible**, which means that Ruby applications can use any existing PostgreSQL client to interact with Materialize as if it were a PostgreSQL database. In this guide, we'll use the  [`pg` gem](https://rubygems.org/gems/pg/) to connect to Materialize and issue PostgreSQL commands.
```
Comment (morsapaes) [inline doc/user/content/guides/ruby.md:10]:
```suggestion
Materialize is **PostgreSQL-compatible**, which means that Ruby applications can use any existing PostgreSQL client to interact with Materialize as if it were a PostgreSQL database. In this guide, we'll use  [`pg` gem](https://rubygems.org/gems/pg/) to connect to Materialize and issue PostgreSQL commands.
```

File: doc/user/content/guides/ruby.md [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+conn = PG.connect(host:"127.0.0.1", port: 6875, user: "materialize")
+
+res  = conn.exec('SELECT * FROM my_view')
+
+res.each do |row|
+  puts row
+end
+```
+
+For more details, see the  [`exec` instancemethod](https://rubydoc.info/gems/pg/0.10.0/PGconn#exec-instance_method) documentation.
```
Comment (morsapaes) [inline doc/user/content/guides/ruby.md:108]:
```suggestion
For more details, see the  [`exec` instance method](https://rubydoc.info/gems/pg/0.10.0/PGconn#exec-instance_method) documentation.
```

**Review by morsapaes — APPROVED** [review]

**Review by ruf-io — APPROVED** [review]

**Pull request merged**